src/components/data/TableRow.js

```diff
@@ -0,0 +1,88 @@
+import classnames from 'classnames';
+import { useContext, useEffect } from 'preact/hooks';
+
+import { downcastRef } from '../../util/typing';
+import { useElementRef } from '../../hooks/use-element-ref';
+
+import ScrollContext from './ScrollContext';
+import TableSectionContext from './TableSectionContext';
+import TableContext from './TableContext';
+
+/**
+ * @typedef {import('../../types').PresentationalProps} CommonProps
+ * @typedef {Omit<import('preact').JSX.HTMLAttributes<HTMLElement>, 'size'>} HTMLAttributes
+ *
+ * @typedef TableRowProps
+ * @prop {boolean} [selected=false]
+ */
+
+/**
+ * Render a table row
+ *
+ * @param {CommonProps & TableRowProps & HTMLAttributes} props
+ */
+const TableRowNext = function TableRow({
+  children,
+  classes,
+  elementRef,
+
+  selected,
+
+  ...htmlAttributes
+}) {
+  const rowRef = useElementRef(elementRef);
+
+  const sectionContext = useContext(TableSectionContext);
+  const scrollContext = useContext(ScrollContext);
+  const tableContext = useContext(TableContext);
+
+  const isHeadRow = sectionContext?.section === 'head';
+
+  // Ensure that a selected row is visible when there is a scrolling context
+  useEffect(() => {
+    if (!selected || !scrollContext || !tableContext) {
+      return;
+    }
+    const scrollEl = scrollContext.scrollRef.current;
+    const tableHead = tableContext.tableRef.current?.querySelector('thead');
+
+    if (selected && scrollEl) {
+      // Ensure the row is visible within the scroll content area
```

Comment:
This comment should be expanded to explain when this can happen: a row is selected on initial render that is outside of the initial scroll viewport (i.e. below it).